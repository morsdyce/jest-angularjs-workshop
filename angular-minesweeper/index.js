import 'common/assets/stylesheets/minefield.css';

import App from 'angular-minesweeper/components/app/app.module';
import Common from 'angular-minesweeper/common/common.module';
import Board from 'angular-minesweeper/components/board/board.module';

export default angular.module('ngReact', [
  Common.name,
  Board.name,
  App.name
]);