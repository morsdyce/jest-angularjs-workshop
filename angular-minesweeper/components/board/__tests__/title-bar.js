import 'angular-minesweeper';

describe('<title-bar>', () => {
  let createComponent, $scope, element;

  beforeEach(angular.mock.module('ngReact.board'));

  beforeEach(inject(($rootScope, $compile) => {
    createComponent = (html, scopeAttrs) => {
      $scope = angular.extend($rootScope.$new(), scopeAttrs);

      element = angular.element(html);
      element = $compile(element)($scope);
      $scope.$digest();
    };
  }));

  test('should render title bar', () => {
    const html = `<title-bar></title-bar>`;
    createComponent(html, {});

    expect(element.html()).toMatchSnapshot();
  });
});

