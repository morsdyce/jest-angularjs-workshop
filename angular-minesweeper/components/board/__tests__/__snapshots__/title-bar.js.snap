// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<title-bar> should render title bar 1`] = `
"<div class=\\"title-bar\\">
  <div class=\\"app-name\\">
    <img ng-src=\\"icon.png\\" src=\\"unsafe:icon.png\\"> Minesweeper
  </div>
  <div class=\\"window-buttons\\">
    <img ng-src=\\"title-buttons.png\\" id=\\"title-buttons\\" src=\\"unsafe:title-buttons.png\\">
  </div>
</div>"
`;
